### PR TITLE
Move ios_content_validation_test to pre-submit tools test

### PR DESCRIPTION
**Pull request by jmagman**
Most of the devicelab task ios_content_validation_test can be moved into the flutter_tool test integration shard.  The archive part of the test needs to stay in the devicelab since it requires codesigning, and the provisioning profiles [aren't set up in LUCI](https://github.com/flutter/flutter/issues/72078).

I reverted two of my own commits this week for post-submit failures in this test that could have been caught in pre-submit.

### CODE REVIEW

File: packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            '-o',
+            '-',
+            infoPlistPath,
+          ],
+        );
+        final bool localNetworkUsageFound = localNetworkUsage.exitCode == 0;
+        expect(localNetworkUsageFound, buildMode == BuildMode.debug);
+      });
+
+      testWithoutContext('check symbols', () {
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart:144]: This check wasn't even in the devicelab test. Bonus!

**Review by jonahwilliams — APPROVED** [review]
LGTM very nice!

File: packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        final ProcessResult grepResult = processManager.runSync(<String>[
+          'grep',
+          '-i',
+          'hello',
+          outputAppFrameworkBinary.path,
+        ]);
+        expect(grepResult.stdout, isNot(contains('matches')));
+      });
+    },
+      skip: !platform.isMacOS,
```
Comment (jmagman) [inline packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart:212]: Hm, this `group` `skip` didn't seem to work, it ran on Windows and Linux.

File: packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+          'vm_snapshot_data',
+        ));
+
+        expect(vmSnapshot.existsSync(), buildMode == BuildMode.debug);
+
+        expect(outputFlutterFramework.childDirectory('Headers'), isNot(exists));
+        expect(outputFlutterFramework.childDirectory('Modules'), isNot(exists));
+
+        // Archiving should contain a bitcode blob, but not building.
+        // This mimics Xcode behavior and present a developer from having to install a
```
Comment (MrBirb) [inline packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart:106]: nit: prevent*

Comment (jmagman) [reply packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart]: Thanks!

File: packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart (hunk 2 above)
Comment (jmagman) [reply packages/flutter_tools/test/integration.shard/ios_content_validation_test.dart]: Oh I know why, it was overridden at the test level https://github.com/dart-lang/test/issues/545

**Comment by jmagman** [comment]
I can't test this, blocked by https://github.com/flutter/flutter/issues/73606

**Pull request merged**